### PR TITLE
refactoring existing documentation

### PR DESCRIPTION
**Pull request by s-radyuk**
EES-3913
In this PR we:
- separated JMeter related docs in separate folder inside docs,
- highlighted the possibility of having different load generators for Kangal

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -2,28 +2,12 @@
...
-1. Save your environment variables in CSV file
-2. Configure test script accordingly. Find detais here [How to write and understand a JMeter test: Test with environment variables](How-to-write-tests.md#test-with-environment-variables)
-3. Add both files in POST request to Kangal API
\ No newline at end of file
+### Load generator types aka backends
+Currently, there are two load generator types implemented for Kangal:
+1. Fake - mock up provider used for testing controller logic. Not generating any load. Useful for debugging.
+2. JMeter - the first real load generator implemented for Kangal. Jmeter load test environments are created by Kangal based on [Kangal-JMeter](https://github.com/hellofresh/kangal-jmeter) docker image. 
+JMeter is a powerful tool which can be used for different performance testing tasks. 
+Please read [JMeter Load generator in Kangal](JMeter in Kangal/JMeter-load-generator-in-kangal.md) for further details.
```
Comment (diegomarangoni) [inline docs/README.md:13]: Those links are not working

File: docs/README.md [hunk 2]
```diff
@@ -1,29 +1,38 @@
...
+- Fake - mock up provider used for testing controller logic. Not generating any load. Useful for debugging.
+
+- JMeter - the first real load generator implemented for Kangal. Kangal creates JMeter load test environments based on [Kangal-JMeter](https://github.com/hellofresh/kangal-jmeter) docker image. 
+JMeter is a powerful tool which can be used for different performance testing tasks. 
+Please read [JMeter Load generator in Kangal](jmeter-in-kangal/JMeter-load-generator-in-kangal.md) for further details.
+
+### Adding a new load generator
+Kangal offers an opportunity to add different load generators as backends. 
+Requirements to new load generators:
+1. Create a docker image that must contain an executable of a new load generator and all required scripts to run it.
```
Comment (aleksandrzhiliaev) [inline docs/README.md:26]:
```suggestion
1. Create a docker image that must contain an executable of a new load generator and all required scripts to run it. Docker image should exit once load test is finished and it should provide logs to stdout which will be used by Kangal Proxy API.
```

File: docs/README.md [hunk 3]
```diff
@@ -1,29 +1,38 @@
...
+2. Create a new backend resource definition in `/pkg/backends/new-backend`.
```
Comment (aleksandrzhiliaev) [inline docs/README.md:27]:
```suggestion
2. Create a new backend resource definition in Kangal source code: 
 - [/pkg/backends](https://github.com/hellofresh/kangal/tree/master/pkg/backends). 
 - [backend.go](https://github.com/hellofresh/kangal/blob/master/pkg/backends/backend.go#L33)
 - [CRD definition](https://github.com/hellofresh/kangal/blob/master/charts/kangal/crd.yaml#L43)
 - [openapi.json](https://github.com/hellofresh/kangal/blob/master/openapi.json#L280)
```

File: docs/README.md [hunk 4]
```diff
@@ -1,29 +1,38 @@
...
+The basic resource is a job that manages all the other resources and sets pods to the `finished` state when the test is over.
+
+## Reporting in Kangal
+Reporting is an important part of load testing process. It basically contains in two parts:
+1. Live metrics during the running load test.
```
Comment (aleksandrzhiliaev) [inline docs/README.md:32]:
```suggestion
1. Live metrics during the running load test - Kangal proxy scrapes logs from main job stdout Docker container.
```

**Comment by iammuho** [comment]
+1

**Comment by aleksandrzhiliaev** [comment]
+1

**Pull request merged**